### PR TITLE
Berry_pots_app_rodata

### PR DESCRIPTION
**Pull request by PikalaxALT**
Extends #289

### CODE REVIEW

**Comment by adrienntindall** [comment]
Address conflicting files

File: src/alph_puzzle.c [hunk 1]
```diff
@@ -434,17 +434,95 @@ static const GraphicsModes sGraphicsMode = {
...
 };
 
-static const BgTemplate sBgTemplate5 = {0, 0, GF_BG_BUF_SIZE_256x256_4BPP, 0x0000, GF_BG_SCR_SIZE_256x256, GX_BG_COLORMODE_16, 30, 0, 0, 2, 0, 0, 0};
+static const BgTemplate sBgTemplate5 = {
+    0, 0,
+    GF_BG_BUF_SIZE_256x256_4BPP,
+    0x000,
+    GF_BG_SCR_SIZE_256x256,
+    GX_BG_COLORMODE_16,
+    30,
```
Comment (adrienntindall) [inline src/alph_puzzle.c:443]: GX_BG consts (if we have them)

File: src/berry_pots_app_tasks.c [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+    args->unk7 = data->unk81;
+    s8 diff = data->unk81 - data->unk80;
+    if (diff > 0) {
+        args->sign = 1;
+    } else {
+        args->sign = -1;
+        diff = -diff;
+    }
+
+    args->unk8 = (s8)(diff * 27) / 4;
```
Comment (adrienntindall) [inline src/berry_pots_app_tasks.c:40]: is s8 cast necessary here / should unk8 be an s8 instead of a u8?

File: src/berry_pots_app_tasks.c [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+    args->unkA = data->unk80;
+    SysTask_CreateOnMainQueue((SysTaskFunc)ov17_022035A4, args, 0);
+    data->runningTasks++;
+}
+
+static void ov17_022035A4(SysTask *task, SysTaskArgs_ov17_0220351C *args) {
+    BerryPotsAppData *unk0 = args->data;
```
Comment (adrienntindall) [inline src/berry_pots_app_tasks.c:47]: unk0 can be named

File: src/berry_pots_app_tasks.c [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+
+    Sprite_AddPositionXY(unk0->sprites[2], args->sign * 4, 0);
+    s16 x, y;
+    Sprite_GetPositionXY(unk0->sprites[2], &x, &y);
+
+    u8 potIndex = (x - 31) / 27;
+    if (potIndex != args->unkA && potIndex != args->unk7) {
+        ov17_02202B58(unk0, potIndex);
+        args->unkA = potIndex;
```
Comment (adrienntindall) [inline src/berry_pots_app_tasks.c:56]: can unkA be documented?

File: src/berry_pots_app_tasks.c [hunk 5]
```diff
@@ -0,0 +1,444 @@
...
+    u32 r0 = ov17_02203B68(data, &flag);
+    if (!flag) {
+        r0 = ov17_02203AD4(data);
+    }
+    return r0;
+}
+
+BOOL ov17_02203A54(BerryPotsAppData *data) {
+    BOOL flag = FALSE;
+    BOOL r0 = ov17_02203C20(data, &flag);
```
Comment (adrienntindall) [inline src/berry_pots_app_tasks.c:265]: r0 -> ret

File: src/berry_pots_app_tasks.c [hunk 6]
```diff
@@ -0,0 +1,444 @@
...
+                return 1;
+            }
+            break;
+    }
+
+    return 2;
+}
+
+static u32 ov17_02203AD4(BerryPotsAppData *data) {
+    BOOL r3 = FALSE;
```
Comment (adrienntindall) [inline src/berry_pots_app_tasks.c:300]: r3 -> flag

File: src/berry_pots_app_tasks.c [hunk 7]
```diff
@@ -0,0 +1,444 @@
...
+        case BERRY_POT_GROWTH_STAGE_NONE:
+            return (unk->mulch != MULCH_NONE) ? 4 : 3;
+        case BERRY_POT_GROWTH_STAGE_PLANTED:
+        case BERRY_POT_GROWTH_STAGE_SPROUTED:
+        case BERRY_POT_GROWTH_STAGE_GROWING:
+        case BERRY_POT_GROWTH_STAGE_BLOOM:
+            return 5;
+        case BERRY_POT_GROWTH_STAGE_BERRIES:
+            return 6;
+
```
Comment (adrienntindall) [inline src/berry_pots_app_tasks.c:439]: extraneous space

File: src/berry_pots_app_tasks.c [hunk 8]
```diff
@@ -0,0 +1,444 @@
...
+        case 0xFF:
```
Comment (adrienntindall) [inline src/berry_pots_app_tasks.c:440]: Make a const for BERRY_POT_GROWTH_STAGE_NONE?

**Review by adrienntindall — CHANGES_REQUESTED** [review]

File: src/berry_pots_app_tasks.c (hunk 8 above)
Comment (PikalaxALT) [reply src/berry_pots_app_tasks.c:440]: 1) that name is already used
2) can't find where growthState can be set to 0xFF

File: src/berry_pots_app_tasks.c (hunk 4 above)
Comment (PikalaxALT) [reply src/berry_pots_app_tasks.c:56]: not yet, would need substantial documentation of this code first

**Review by adrienntindall — APPROVED** [review]

**Pull request merged**